src/realm/sync/client.cpp

```diff
@@ -1103,24 +1099,53 @@ SyncClientHookAction SessionImpl::call_debug_hook(SyncClientHookEvent event, con
     return call_debug_hook(data);
 }
 
-bool SessionImpl::is_steady_state_download_message(DownloadBatchState batch_state, int64_t query_version)
+bool SessionImpl::is_steady_state_download_message(DownloadBatchState batch_state, int64_t query_version,
```

Comment:
pretty convoluted. I have an approach bellow where this method should not be needed at all anymore.

Reply:
@jbreams had some suggestions on a [separate commit](https://github.com/realm/realm-core/commit/b6e0c6093f5ae030fbadc0540bd68d9ba4f41243) that reworked this function and the tracking of the changeset remote versions. This has been integrated into this PR and the tests have been updated to validate the different paths depending on how the changeset looks.